gstools/field/base.py

```diff
@@ -26,33 +27,264 @@
+        # mesh_type needs a special setter, therefore, `set_field_data` is not
+        # used here
+        self.mesh_type = mesh_type
+
+        # the pos/ points of the mesh
+        self._pos = pos
+
+        # data stored at each pos/ point, the "fields"
+        if values is not None:
+            self.point_data = {name: values}
+        else:
+            self.point_data = {}
+
+        # data valid for the global field
+        self.field_data = {}
+
+        self.set_field_data("default_field", name)
+
+        self.field_data["mesh_type"] = mesh_type
+
+    def set_field_data(self, name, value):
+        """Add an attribute to this instance and add it the `field_data`
+
+        This helper method is used to create attributes for easy access
+        while using this class, but it also adds an entry to the dictionary
+        `field_data`, which is used for exporting the data.
+        """
+        setattr(self, name, value)
+        self.field_data[name] = value
+
+    def add_field(
+        self, values, name="field", is_default_field=False,
+    ):
+        """Add a field (point data) to the mesh
+
+        .. note::
+            If no field has existed before calling this method,
+            the given field will be set to the default field.
+
+        .. warning::
+            If point data with the same `name` already exists, it will be
+            overwritten.
+
+        Parameters
+        ----------
+        values : :class:`numpy.ndarray`
+            the point data, has to be the same shape as the mesh
+        name : :class:`str`, optional
+            the name of the point data
+        is_default_field : :class:`bool`, optional
+            is this the default field of the mesh?
+
+        """
+        values = np.array(values)
+        self._check_point_data(values)
+        self.point_data[name] = values
+        # set the default_field to the first field added
+        if len(self.point_data) == 1 or is_default_field:
+            self.set_field_data("default_field", name)
+
+    def __getitem__(self, key):
+        """:any:`numpy.ndarray`: The values of the field."""
+        return self.point_data[key]
+
+    def __setitem__(self, key, value):
+        self.point_data[key] = value
+
+    @property
+    def pos(self):
+        """:any:`numpy.ndarray`: The pos. on which the field is defined."""
+        return self._pos
+
+    @pos.setter
+    def pos(self, value):
+        """
+        Warning: setting new positions deletes all previously stored fields.
+        """
+        self.point_data = {self.default_field: None}
+        self._pos = value
+
+    @property
+    def field(self):
+        """:class:`numpy.ndarray`: The point data of the default field."""
+        return self.point_data[self.default_field]
+
+    @field.setter
+    def field(self, values):
+        self._check_point_data(values)
+        self.point_data[self.default_field] = values
+
+    @property
+    def value_type(self, field="field"):
```

Comment:
You shouldn't have any arguments in the property accessor - its not usable by the end-user. perhaps this needs to be an standard method as right now it will always have `"field"` as the argument for `field`

Reply:
Good point :-D
Don't know how that happened.